scalableminds/webknossos
# Allow to extrude segment instead of interpolate (similar to the old "copy from previous slice" feature)

This PR generalizes the volume interpolation feature to also provide an "extrusion" feature (i.e., copy a segment from one slice to N slices). This is similar to the old copy-from-previous slice feature, but a bit more powerful (since one can label N slices at the same time).
wK remembers whether interpolation or extrusion was used last and switches the main button accordingly.

Screenshots:
![image](https://user-images.githubusercontent.com/2486553/182850194-4e17a206-c531-477a-af85-447c7077e40e.png)

![image](https://user-images.githubusercontent.com/2486553/182850108-37ee8255-5af2-42cd-bf74-5ae1405c20d5.png)

![image](https://user-images.githubusercontent.com/2486553/182850142-2514986a-a996-4e9e-9819-ff0b2ba3f430.png)


### URL of deployed dev instance (used for testing):
- https://restorecopyfromslice.webknossos.xyz/

### Steps to test:
- use volume interpolation as usual (via shortcut V or via button in the toolbar)
- label a slice, move a few slices via `f` and then use the extrusion feature via the dropdown in the toolbar
- the segment should be copied to all slices between the start and the current slice (including the current slice)
- also, the button should change to the extrusion icon
- using the feature again with the extrusion button (without the dropdown) should also work as expected

### Issues:
- fixes #6338 

------
(Please delete unneeded items, merge only when none are left open)
